[Style] Adopt strong style types for properties using FixedVector

weinig opened the pull request:
#### a26105eb6359c033c186c2f35398be2e7ab6363f
<pre>
[Style] Adopt strong style types for properties using FixedVector
<a href="https://bugs.webkit.org/show_bug.cgi?id=295291">https://bugs.webkit.org/show_bug.cgi?id=295291</a>

Reviewed by Darin Adler.

Converts most of the remaining list properties to strong style types.

- Adds new aggregates SpaceSeparatedFixedVector and CommaSeparatedFixedVector
  to compliment the existing SpaceSeparatedVector and CommaSeparatedVector but
  for FixedVector.
- Beefs up ListOrNone adaptor to allow basic list operations.
- Adds a new adaptor, ListOrDefault, which rather than using the empty list
  value for `none`, allows clients to specify a &quot;default&quot; value to use when
  the list is empty. This means it can never actually appear empty. Useful
  for properties like `scroll-timeline-axis` where they defined to never be
  empty.
- Adds support for automatic conversion from CSSValue to new FixedVector
  aggregates and ListOr* adaptors.
- Makes WTF::makeReversedRange reflect whether the provided range had a
  `size()` function, allowing for more efficient uses, such as when provided
  to FixedVector::map.

* Source/WTF/wtf/IteratorRange.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/CSSAnimation.cpp:
* Source/WebCore/animation/ScrollTimeline.cpp:
* Source/WebCore/animation/ScrollTimeline.h:
* Source/WebCore/animation/StyleOriginatedTimelinesController.cpp:
* Source/WebCore/animation/StyleOriginatedTimelinesController.h:
* Source/WebCore/animation/ViewTimeline.cpp:
* Source/WebCore/animation/ViewTimeline.h:
* Source/WebCore/animation/WebAnimationTypes.cpp: Added.
* Source/WebCore/animation/WebAnimationTypes.h:
* Source/WebCore/animation/WebAnimationUtilities.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/values/CSSValueAggregates.h:
* Source/WebCore/css/values/filter-effects/CSSAppleColorFilterProperty.h:
* Source/WebCore/css/values/filter-effects/CSSFilterProperty.h:
* Source/WebCore/dom/ViewTransition.cpp:
* Source/WebCore/editing/Editor.cpp:
* Source/WebCore/platform/animation/Animation.h:
* Source/WebCore/rendering/EllipsisBoxPainter.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderTheme.cpp:
* Source/WebCore/rendering/StyledMarkedText.cpp:
* Source/WebCore/rendering/StyledMarkedText.h:
* Source/WebCore/rendering/TextBoxPainter.cpp:
* Source/WebCore/rendering/TextDecorationPainter.cpp:
* Source/WebCore/rendering/TextDecorationPainter.h:
* Source/WebCore/rendering/TextPaintStyle.cpp:
* Source/WebCore/rendering/TextPaintStyle.h:
* Source/WebCore/rendering/TextPainter.cpp:
* Source/WebCore/rendering/TextPainter.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
* Source/WebCore/rendering/style/StyleMiscNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleMiscNonInheritedData.h:
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareInheritedData.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/rendering/svg/RenderSVGBlock.cpp:
* Source/WebCore/rendering/svg/RenderSVGText.cpp:
* Source/WebCore/rendering/svg/SVGTextBoxPainter.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGModelObject.cpp:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/ScopedName.cpp: Added.
* Source/WebCore/style/ScopedName.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/Styleable.cpp:
* Source/WebCore/style/values/StyleValueTypes+CSSValueConversion.h: Added.
* Source/WebCore/style/values/anchor-position/StyleAnchorName.h: Added.
* Source/WebCore/style/values/borders/StyleBoxShadow.cpp:
* Source/WebCore/style/values/borders/StyleBoxShadow.h:
* Source/WebCore/style/values/borders/StyleShadow.h:
* Source/WebCore/style/values/contain/StyleContainerName.h: Added.
* Source/WebCore/style/values/primitives/StylePrimitiveKeyword+CSSValueConversion.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleProgressTimelineAxes.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleProgressTimelineName.cpp: Added.
* Source/WebCore/style/values/scroll-animations/StyleProgressTimelineName.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleScrollTimelines.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleViewTimelineInsets.cpp: Added.
* Source/WebCore/style/values/scroll-animations/StyleViewTimelineInsets.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleViewTimelines.h: Added.
* Source/WebCore/style/values/text-decoration/StyleTextShadow.cpp:
* Source/WebCore/style/values/text-decoration/StyleTextShadow.h:
* Source/WebCore/style/values/view-transitions/StyleViewTransitionClass.h: Added.

Canonical link: <a href="https://commits.webkit.org/297032@main">https://commits.webkit.org/297032@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3471bc1b445e960c76fb4a6e94e4efa1cb34f55b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110347 "59 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30006 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20438 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116371 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60599 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112310 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30685 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38593 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83902 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113295 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24481 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99362 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64347 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23840 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17501 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60167 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/102838 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93860 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17557 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119161 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/108901 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37387 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27728 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92870 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37759 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95630 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92696 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23615 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37696 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15428 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33293 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37281 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42752 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/133176 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36944 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35997 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/jitless-wasm-to-js-should-follow-to-wasm-call-frame-convention.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40283 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38652 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/animation/ScrollTimeline.cpp:
```diff
@@ -388,14 +388,9 @@ void ScrollTimeline::animationTimingDidChange(WebAnimation& animation)
 }
 
-TextStream& operator<<(TextStream& ts, Scroller scroller)
+TextStream& operator<<(TextStream& ts, const ScrollTimeline& scrollTimeline)
```
```suggestion
TextStream& operator<<(TextStream& ts, const ScrollTimeline& timeline)
```

darinadler commented on Source/WebCore/animation/ScrollTimeline.cpp:
```diff
@@ -388,14 +388,9 @@ void ScrollTimeline::animationTimingDidChange(WebAnimation& animation)
-    case Scroller::Self: ts << "self"_s; break;
-    }
-    return ts;
+    return ts << scrollTimeline.name() << ' ' << scrollTimeline.axis();
```
```suggestion
    return ts << timeline.name() << ' ' << timeline.axis();
```

darinadler commented on Source/WebCore/animation/ViewTimeline.cpp:
```diff
@@ -639,4 +639,9 @@ WTF::TextStream& operator<<(WTF::TextStream& ts, const StickinessAdjustmentData:
+TextStream& operator<<(TextStream& ts, const ViewTimeline& viewTimeline)
+{
+    return ts << viewTimeline.name() << ' ' << viewTimeline.axis() << ' ' << viewTimeline.insets();
+}
```
```suggestion
TextStream& operator<<(TextStream& ts, const ViewTimeline& timeline)
{
    return ts << timeline.name() << ' ' << timeline.axis() << ' ' << timeline.insets();
}
```

darinadler commented on Source/WebCore/animation/ViewTimeline.h:
```diff
@@ -71,15 +71,15 @@ struct StickinessAdjustmentData {
 
-    const ViewTimelineInsets& insets() const { return m_insets; }
-    void setInsets(ViewTimelineInsets&& insets) { m_insets = WTFMove(insets); }
+    const ViewTimelineInsetItem& insets() const { return m_insets; }
```
Why is the type name singular and the function name plural?

weinig replied:
Function should be renamed to insetItem I think.

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
 
+// Wraps a variable (though known at construction) number of elements of a single type, semantically marking them as serializing as "space separated".
+template<typename T> struct SpaceSeparatedFixedVector {
+    using FixedVector = WTF::FixedVector<T>;
```
Would it be too confusing to have this member type be named `Vector` instead? Or maybe `Container`?

weinig replied:
Container is better for sure. Or List like I have below.

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+    template<typename SizedRange, typename Mapper>
+    static SpaceSeparatedFixedVector<T> map(SizedRange&& range, NOESCAPE Mapper&& mapper)
+    {
+        return WTF::FixedVector<T>::map(std::forward<SizedRange>(range), std::forward<Mapper>(mapper));
```
```suggestion
        return FixedVector::map(std::forward<SizedRange>(range), std::forward<Mapper>(mapper));
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+    {
+    }
+
+    SpaceSeparatedFixedVector(WTF::FixedVector<T>&& value)
```
Consider this?
```suggestion
    SpaceSeparatedFixedVector(FixedVector&& value)
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+
+    template<typename F> decltype(auto) map(F&& functor) const { return value.map(std::forward<F>(functor)); }
+
+    bool operator==(const SpaceSeparatedFixedVector<T>&) const = default;
```
```suggestion
    bool operator==(const SpaceSeparatedFixedVector&) const = default;
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+    {
+    }
+
+    CommaSeparatedFixedVector(WTF::FixedVector<T>&& value)
```
Consider this?
```suggestion
    CommaSeparatedFixedVector(FixedVector&& value)
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+
+    template<typename F> decltype(auto) map(F&& functor) const { return value.map(std::forward<F>(functor)); }
+
+    bool operator==(const CommaSeparatedFixedVector<T>&) const = default;
```
```suggestion
    bool operator==(const CommaSeparatedFixedVector&) const = default;
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+    }
+
+    template<typename SizedRange, typename Mapper>
+    static SpaceSeparatedFixedVector<T> map(SizedRange&& range, NOESCAPE Mapper&& mapper)
```
```suggestion
    static SpaceSeparatedFixedVector map(SizedRange&& range, NOESCAPE Mapper&& mapper)
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+    }
+
+    template<typename SizedRange, typename Mapper>
+    static CommaSeparatedFixedVector<T> map(SizedRange&& range, NOESCAPE Mapper&& mapper)
```
```suggestion
    static CommaSeparatedFixedVector map(SizedRange&& range, NOESCAPE Mapper&& mapper)
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+    template<typename SizedRange, typename Mapper>
+    static CommaSeparatedFixedVector<T> map(SizedRange&& range, NOESCAPE Mapper&& mapper)
+    {
+        return WTF::FixedVector<T>::map(std::forward<SizedRange>(range), std::forward<Mapper>(mapper));
```
```suggestion
        return FixedVector::map(std::forward<SizedRange>(range), std::forward<Mapper>(mapper));
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+
+    bool operator==(const CommaSeparatedFixedVector<T>&) const = default;
+
+    WTF::FixedVector<T> value;
```
```suggestion
    FixedVector value;
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -267,25 +268,153 @@ template<typename T, size_t inlineCapacity = 0> struct CommaSeparatedVector {
+
+    bool operator==(const SpaceSeparatedFixedVector<T>&) const = default;
+
+    WTF::FixedVector<T> value;
```
```suggestion
    FixedVector value;
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -304,6 +433,99 @@ template<typename T> struct ListOrNone {
+            if (owner->isDefault())
+                return owner->defaulter();
+            else
+                return *it;
```
```suggestion
            return *it;
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -304,6 +433,99 @@ template<typename T> struct ListOrNone {
+        const_iterator operator++(int)
+        {
+            auto result = *this;
+            ++(*this);
```
Maybe this?
```suggestion
            ++*this;
```

darinadler commented on Source/WebCore/css/values/CSSValueAggregates.h:
```diff
@@ -304,6 +433,99 @@ template<typename T> struct ListOrNone {
+        {
+            if (owner->isDefault()) {
+                if (!atEndForDefault)
+                    atEndForDefault = true;
```
```suggestion
                atEndForDefault = true;
```

darinadler commented on Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
```diff
@@ -217,14 +213,14 @@ class StyleRareNonInheritedData : public RefCounted<StyleRareNonInheritedData> {
-    FixedVector<ViewTimelineInsets> viewTimelineInsets;
-    FixedVector<AtomString> viewTimelineNames;
+    Style::ViewTimelines viewTimelines;
+    Style::ViewTimelineInsets viewTimelineInsets;
+    Style::ProgressTimelineAxes viewTimelineAxes;
```
Why no longer sorted alphabetically? Is there a different rationale for the order here?

weinig replied:
I was keeping the Progress prefixed items together, but I don’t think it was the right call and will put it back to the old order.

darinadler commented on Source/WebCore/style/values/scroll-animations/StyleProgressTimelineName.cpp:
```diff
@@ -0,0 +1,49 @@
+
+#include "CSSPrimitiveValue.h"
+#include "StyleBuilderChecking.h"
+#include <wtf/NeverDestroyed.h>
```
I don't see this used here.

darinadler commented on Source/WebCore/style/ScopedName.h:
```diff
@@ -36,14 +36,29 @@ struct ScopedName {
 } // namespace Style
 } // namespace WebCore
+
+template<> inline constexpr auto WebCore::TreatAsVariantLike<WebCore::Style::ScopedName> = true;
```
Why isn't this inside the WebCore namespace?

darinadler replied:
I also think that `constexpr` in cases like this is equivalent to `inline constexpr`.

weinig replied:
For constants like this, `inline constexpr` is necessary, otherwise it can lead to ODR violations. (I used to have a good blog post to link about this, but I can't find it right now. https://quuxplusone.github.io/blog/2022/07/08/inline-constexpr/ does an ok job explaining it).

weinig committed:
[Style] Adopt strong style types for properties using FixedVector
https://bugs.webkit.org/show_bug.cgi?id=295291

Reviewed by Darin Adler.

Converts most of the remaining list properties to strong style types.

- Adds new aggregates SpaceSeparatedFixedVector and CommaSeparatedFixedVector
  to compliment the existing SpaceSeparatedVector and CommaSeparatedVector but
  for FixedVector.
- Beefs up ListOrNone adaptor to allow basic list operations.
- Adds a new adaptor, ListOrDefault, which rather than using the empty list
  value for `none`, allows clients to specify a "default" value to use when
  the list is empty. This means it can never actually appear empty. Useful
  for properties like `scroll-timeline-axis` where they defined to never be
  empty.
- Adds support for automatic conversion from CSSValue to new FixedVector
  aggregates and ListOr* adaptors.
- Makes WTF::makeReversedRange reflect whether the provided range had a
  `size()` function, allowing for more efficient uses, such as when provided
  to FixedVector::map.

* Source/WTF/wtf/IteratorRange.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/CSSAnimation.cpp:
* Source/WebCore/animation/ScrollTimeline.cpp:
* Source/WebCore/animation/ScrollTimeline.h:
* Source/WebCore/animation/StyleOriginatedTimelinesController.cpp:
* Source/WebCore/animation/StyleOriginatedTimelinesController.h:
* Source/WebCore/animation/ViewTimeline.cpp:
* Source/WebCore/animation/ViewTimeline.h:
* Source/WebCore/animation/WebAnimationTypes.cpp: Added.
* Source/WebCore/animation/WebAnimationTypes.h:
* Source/WebCore/animation/WebAnimationUtilities.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/values/CSSValueAggregates.h:
* Source/WebCore/css/values/filter-effects/CSSAppleColorFilterProperty.h:
* Source/WebCore/css/values/filter-effects/CSSFilterProperty.h:
* Source/WebCore/dom/ViewTransition.cpp:
* Source/WebCore/editing/Editor.cpp:
* Source/WebCore/platform/animation/Animation.h:
* Source/WebCore/rendering/EllipsisBoxPainter.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderTheme.cpp:
* Source/WebCore/rendering/StyledMarkedText.cpp:
* Source/WebCore/rendering/StyledMarkedText.h:
* Source/WebCore/rendering/TextBoxPainter.cpp:
* Source/WebCore/rendering/TextDecorationPainter.cpp:
* Source/WebCore/rendering/TextDecorationPainter.h:
* Source/WebCore/rendering/TextPaintStyle.cpp:
* Source/WebCore/rendering/TextPaintStyle.h:
* Source/WebCore/rendering/TextPainter.cpp:
* Source/WebCore/rendering/TextPainter.h:
* Source/WebCore/rendering/cocoa/RenderThemeCocoa.mm:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.cpp:
* Source/WebCore/rendering/style/SVGRenderStyleDefs.h:
* Source/WebCore/rendering/style/StyleMiscNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleMiscNonInheritedData.h:
* Source/WebCore/rendering/style/StyleRareInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareInheritedData.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/rendering/svg/RenderSVGBlock.cpp:
* Source/WebCore/rendering/svg/RenderSVGText.cpp:
* Source/WebCore/rendering/svg/SVGTextBoxPainter.cpp:
* Source/WebCore/rendering/svg/legacy/LegacyRenderSVGModelObject.cpp:
* Source/WebCore/style/AnchorPositionEvaluator.cpp:
* Source/WebCore/style/ScopedName.cpp: Added.
* Source/WebCore/style/ScopedName.h:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/Styleable.cpp:
* Source/WebCore/style/values/StyleValueTypes+CSSValueConversion.h: Added.
* Source/WebCore/style/values/anchor-position/StyleAnchorName.h: Added.
* Source/WebCore/style/values/borders/StyleBoxShadow.cpp:
* Source/WebCore/style/values/borders/StyleBoxShadow.h:
* Source/WebCore/style/values/borders/StyleShadow.h:
* Source/WebCore/style/values/contain/StyleContainerName.h: Added.
* Source/WebCore/style/values/primitives/StylePrimitiveKeyword+CSSValueConversion.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleProgressTimelineAxes.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleProgressTimelineName.cpp: Added.
* Source/WebCore/style/values/scroll-animations/StyleProgressTimelineName.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleScrollTimelines.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleViewTimelineInsets.cpp: Added.
* Source/WebCore/style/values/scroll-animations/StyleViewTimelineInsets.h: Added.
* Source/WebCore/style/values/scroll-animations/StyleViewTimelines.h: Added.
* Source/WebCore/style/values/text-decoration/StyleTextShadow.cpp:
* Source/WebCore/style/values/text-decoration/StyleTextShadow.h:
* Source/WebCore/style/values/view-transitions/StyleViewTransitionClass.h: Added.

Canonical link: https://commits.webkit.org/297032@main
